<?xml version="1.0"?>
<launch>

    <arg name="vocabularty_path" default="$(find ORB_SLAM2_CUDA)/../../../Vocabulary/ORBvoc.txt" />
    <arg name="camera_setting_path" default="$(find ORB_SLAM2_CUDA)/conf/Spiri_lowRes_Stereo.yaml" />
    <arg name="bUseViewer" default="false" />
    <arg name="bEnablePublishROSTopic" default="true" />
    <arg name="do_rectify" default="false" />

    <node name="ORB_SLAM2_CUDA" pkg="ORB_SLAM2_CUDA" type="Stereo" output="screen" 
    	args="$(arg vocabularty_path) $(arg camera_setting_path) $(arg bUseViewer) $(arg bEnablePublishROSTopic) $(arg do_rectify) " >
        <remap from="/camera/left/image_raw" to="/stereo/left/image_rect" />
	<remap from="/camera/right/image_raw" to="/stereo/right/image_rect" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="world_to_local_origin" args="0 0 0 0 0 0 world local_origin 10" />
    <!--<node name="vision_odom_transformer" pkg="topic_tools" type="transform" output="screen" args="/posestamped_vision /mavros/vision_pose/pose geometry_msgs/PoseStamped 'geometry_msgs.msg.PoseStamped(header=std_msgs.msg.Header(seq=m.header.seq,stamp=m.header.stamp,frame_id=&quot;local_origin&quot;),pose=geometry_msgs.msg.Pose(position=geometry_msgs.msg.Point(x=m.pose.position.z * 1, y=m.pose.position.x * 1, z=m.pose.position.y * -1),orientation = m.pose.orientation))' import geometry_msgs std_msgs" />-->
    <node name="vision_odom_transformer" pkg="topic_tools" type="transform" output="screen" args="/posestamped_vision /mavros/vision_pose/pose geometry_msgs/PoseStamped 'geometry_msgs.msg.PoseStamped(header=std_msgs.msg.Header(seq=m.header.seq,stamp=m.header.stamp,frame_id=&quot;local_origin&quot;),pose=geometry_msgs.msg.Pose(position=geometry_msgs.msg.Point(x=m.pose.position.x * 1, y=m.pose.position.y * 1, z=m.pose.position.z * 1),orientation = m.pose.orientation))' --import geometry_msgs std_msgs" />
</launch>
